homeassistant/components/fritz/binary_sensor.py

```diff
@@ -0,0 +1,100 @@
+"""AVM FRITZ!Box connectivitiy sensor."""
+import logging
+
+from fritzconnection.core.exceptions import FritzConnectionException
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_CONNECTIVITY,
+    BinarySensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up entry."""
+    _LOGGER.debug("Setting up FRITZ!Box binary sensors")
+    fritzbox_tools = hass.data[DOMAIN][entry.entry_id]
+
+    if "WANIPConn1" in fritzbox_tools.connection.services:
+        # Only routers are supported at the moment
+        async_add_entities(
+            [FritzBoxConnectivitySensor(fritzbox_tools, entry.title)], True
+        )
+
+    return True
+
+
+class FritzBoxConnectivitySensor(BinarySensorEntity):
+    """Define FRITZ!Box connectivity class."""
+
+    def __init__(self, fritzbox_tools, device_friendlyname: str):
+        """Init FRITZ!Box connectivity class."""
+        self._fritzbox_tools = fritzbox_tools
+        self._unique_id = f"{self._fritzbox_tools.unique_id}-connectivity"
+        self._name = f"{device_friendlyname} Connectivity"
+        self._is_on = True
+        self._is_available = True
+        self._attributes = {}
+        super().__init__()
+
+    @property
+    def name(self):
+        """Return name."""
+        return self._name
+
+    @property
+    def device_class(self):
+        """Return device class."""
+        return DEVICE_CLASS_CONNECTIVITY
+
+    @property
+    def is_on(self) -> bool:
+        """Return status."""
+        return self._is_on
+
+    @property
+    def unique_id(self):
+        """Return unique id."""
+        return self._unique_id
+
+    @property
+    def device_info(self):
+        """Return device info."""
+        return self._fritzbox_tools.device_info
+
+    @property
+    def available(self) -> bool:
+        """Return availability."""
+        return self._is_available
+
+    @property
+    def device_state_attributes(self) -> dict:
+        """Return device attributes."""
+        return self._attributes
+
```

Comment:
Since this entity is part of the device you are setting up in `async_device_setup`, you can set the `device_info` here instead.  Since you plan on multiple entities, make another base class in `common.py` ..maybe `FritzBoxHostEntity` and put the `device_info` there so all the new entities can inherit that base class and you won't have to write it multiple times.

Then you can add `via_device` to `device_info` in `device_tracker.py`

Reply:
Fixed